tests/jobs/ahm_analysis/test_integration.py

```diff
@@ -0,0 +1,102 @@
+from res.enkf.export import (
+    SummaryObservationCollector,
+    GenDataObservationCollector,
+)
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_ahmanalysis_run(test_data_root):
+    """test data_set with only scalar parameters"""
+    test_data_dir = os.path.join(test_data_root, "snake_oil")
+
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("snake_oil.ert")
+    res_config.convertToCReference(None)
+    ert = EnKFMain(res_config)
+
+    ahmanalysis.AhmAnalysisJob(ert).run(prior_name="default_0")
+
+    # assert that this returns/generates a KS csv file
+    output_ks = os.path.join(
+        "storage", "snake_oil", "reports", "default_0", "AhmAnalysisJob", "ks.csv"
+    )
+    assert os.path.isfile(output_ks)
+    ks_df = pd.read_csv(output_ks)
+    assert not ks_df.empty
+    output_activeobs = output_ks.replace("ks.csv", "active_obs_info.csv")
+    assert os.path.isfile(output_activeobs)
+    output_misfitobs = output_ks.replace("ks.csv", "misfit_obs_info.csv")
+    assert os.path.isfile(output_misfitobs)
+    output_prior = output_ks.replace("ks.csv", "prior.csv")
+    assert os.path.isfile(output_prior)
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_ahmanalysis_run_field(test_data_root, grid_prop):
+    """test data_set with scalar and Field parameters"""
+    test_data_dir = os.path.join(test_data_root, "snake_oil")
+
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+    os.makedirs("fields")
+    grid = EclGridGenerator.createRectangular((10, 12, 5), (1, 1, 1))
+    for iens in range(10):
+        grid_prop("PERMX", 10, grid.getGlobalSize(), "fields/permx%d.grdecl" % iens)
+        grid_prop("PORO", 0.2, grid.getGlobalSize(), "fields/poro%d.grdecl" % iens)
+
+    res_config = ResConfig("snake_oil_field.ert")
+    res_config.convertToCReference(None)
+    ert = EnKFMain(res_config)
+
+    ahmanalysis.AhmAnalysisJob(ert).run(prior_name="default")
+
+    # assert that this returns/generates the delta field parameter
+    gen_obs_list = GenDataObservationCollector.getAllObservationKeys(ert)
+    summary_obs_list = SummaryObservationCollector.getAllObservationKeys(ert)
+    obs_keys = gen_obs_list + summary_obs_list
+    output_deltafield = os.path.join(
+        "storage",
+        "snake_oil_field",
+        "reports",
+        "default",
+        "AhmAnalysisJob",
+        "delta_fieldPERMX.csv",
+    )
+    assert os.path.isfile(output_deltafield)
+    delta_df = pd.read_csv(output_deltafield, index_col=0)
+    assert len(delta_df.columns) == 8 + (len(obs_keys) * 2) + 1
+    # check field parameter is present and not empty in the final KS matrix
+    output_ks = os.path.join(
+        "storage", "snake_oil_field", "reports", "default", "AhmAnalysisJob", "ks.csv"
+    )
+    ks_df = pd.read_csv(output_ks, index_col=0)
+    assert not ks_df.empty
```

Comment:
Would like to have some more concrete checks than this, to make sure that the script does the right thing

Reply:
I have added some more concrete test on min/max Ks value and expected groups of observations in the KS dataframe, Is it enough ?